### PR TITLE
Create meson_cross_file.txt for cross compilation

### PR DESCRIPTION
**Pull request by h-vetinari**
Based on what exists in https://github.com/conda-forge/clang-compiler-activation-feedstock + https://github.com/conda-forge/clang-compiler-activation-feedstock/pull/100, in order to solve cases like https://github.com/conda-forge/scipy-feedstock/pull/205; numpy, scipy & pandas are all moving to meson, so we'll need to sort this out eventually.

I admit that I don't know the ins & outs of this feedstock, it's very possible that this needs further work.

### CODE REVIEW

**Comment by isuruf** [comment]
You should also add https://github.com/conda-forge/clang-compiler-activation-feedstock/blob/a9f5382dbd3d753aa0a8d27a8c0575c6ee4ea363/recipe/activate-clang.sh#L145

File: recipe/activate-gcc.sh [hunk 1]
```diff
@@ -148,9 +148,23 @@ if [ "${CONDA_BUILD:-0}" = "1" ]; then
   _CMAKE_ARGS="${_CMAKE_ARGS} -DCMAKE_PROGRAM_PATH=${BUILD_PREFIX}/bin;$PREFIX/bin"
 fi
 
+_MESON_ARGS="--buildtype release"
+
 # shellcheck disable=SC2050 # templating will fix this error
 if [ "@CONDA_BUILD_CROSS_COMPILATION@" = "1" ]; then
   _CMAKE_ARGS="${_CMAKE_ARGS} -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_PROCESSOR=@LINUX_MACHINE@"
+  _MESON_ARGS="${_MESON_ARGS} --prefix="$PREFIX" -Dlibdir=lib"
```
Comment (isuruf) [inline recipe/activate-gcc.sh:156]: This should be in the if block starting at Line 144

Comment (h-vetinari) [reply recipe/activate-gcc.sh]: Sorry for the oversight and thanks for the attentiveness.

File: recipe/activate-gcc.sh [hunk 2]
```diff
@@ -146,11 +146,25 @@ if [ "${CONDA_BUILD:-0}" = "1" ]; then
   _CMAKE_ARGS="${_CMAKE_ARGS} -DCMAKE_FIND_ROOT_PATH=$PREFIX;${BUILD_PREFIX}/@CHOST@/sysroot"
   _CMAKE_ARGS="${_CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=${PREFIX} -DCMAKE_INSTALL_LIBDIR=lib"
   _CMAKE_ARGS="${_CMAKE_ARGS} -DCMAKE_PROGRAM_PATH=${BUILD_PREFIX}/bin;$PREFIX/bin"
+  _MESON_ARGS="${_MESON_ARGS} --prefix=$PREFIX -Dlibdir=lib"
 fi
 
+_MESON_ARGS="--buildtype release"
+
```
Comment (isuruf) [inline recipe/activate-gcc.sh:153]: This should go before Line 144

**Comment by isuruf** [comment]
Thanks

**Pull request merged**